xarray/tests/test_treenode.py

```diff
@@ -337,12 +320,12 @@ def test_descendants(self):
         descendants = root.descendants
         expected = [
             "b",
+            "c",
```

Comment:
With the values of `expected`, I'm guessing that because it's all just built-in Python types (a `list` or `str` values) `mypy` can understand that ([here is a random similar example from test_dataset.py](https://github.com/pydata/xarray/blob/main/xarray/tests/test_dataset.py#L3481).

For the places where `create_test_tree` is called (so `root` in this test), the typing is done within that test function, so my guess is that `mypy` is pulling the typing from there (see L242 - L250).

Reply:
Let's leave worrying about mypy on tests for #8926